azurerm/resource_arm_private_link_endpoint.go

```diff
@@ -0,0 +1,332 @@
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	privateServiceConnections := d.Get("private_service_connection").([]interface{})
+	subnetId := d.Get("subnet_id").(string)
+	t := d.Get("tags").(map[string]interface{})
+
+	parameters := network.PrivateEndpoint{
+		Location: utils.String(location),
+		PrivateEndpointProperties: &network.PrivateEndpointProperties{
+			PrivateLinkServiceConnections:       expandArmPrivateLinkEndpointServiceConnection(privateServiceConnections, false),
+			ManualPrivateLinkServiceConnections: expandArmPrivateLinkEndpointServiceConnection(privateServiceConnections, true),
+			Subnet: &network.Subnet{
+				ID: utils.String(subnetId),
+			},
+		},
+		Tags: tags.Expand(t),
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error creating Private Link Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of Private Link Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		return fmt.Errorf("Error retrieving Private Link Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("API returns a nil/empty id on Private Link Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmPrivateLinkEndpointRead(d, meta)
+}
+
+func resourceArmPrivateLinkEndpointRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Network.PrivateEndpointClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["privateEndpoints"]
+
+	resp, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Private Link Endpoint %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Private Link Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if props := resp.PrivateEndpointProperties; props != nil {
+		if subnet := props.Subnet; subnet != nil {
+			d.Set("subnet_id", subnet.ID)
+		}
+		if err := d.Set("private_service_connection", flattenArmPrivateLinkEndpointServiceConnection(props.PrivateLinkServiceConnections, props.ManualPrivateLinkServiceConnections)); err != nil {
+			return fmt.Errorf("Error setting `private_service_connection`: %+v", err)
+		}
+		if props.NetworkInterfaces != nil {
```

Comment:
the nil check should be inside the flatten function

Reply:
Fixed.

Reply:
This check should be in the flatten function